definitions/features/sync_plans.rb

```diff
@@ -47,6 +47,7 @@ def update_records(ids, enabled)
   ensure
     new_data = sync_plan_data(enabled, updated_record_ids)
     save_state(new_data)
+    updated_record_ids
```

Comment:
As we already talked in a past: ensure block will not influence what the method returns. Please move it before the ensure block